homeassistant/components/androidtv/media_player.py

```diff
@@ -220,40 +240,63 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
-
-    if hass.services.has_service(ANDROIDTV_DOMAIN, SERVICE_ADB_COMMAND):
-        return
-
-    def service_adb_command(service):
-        """Dispatch service calls to target entities."""
-        cmd = service.data.get(ATTR_COMMAND)
-        entity_id = service.data.get(ATTR_ENTITY_ID)
-        target_devices = [
-            dev
-            for dev in hass.data[ANDROIDTV_DOMAIN].values()
-            if dev.entity_id in entity_id
-        ]
-
-        for target_device in target_devices:
-            output = target_device.adb_command(cmd)
-
-            # log the output, if there is any
-            if output:
-                _LOGGER.info(
-                    "Output of command '%s' from '%s': %s",
-                    cmd,
-                    target_device.entity_id,
-                    output,
-                )
+        _LOGGER.debug("Setup %s at %s %s", device_name, address, adb_log)
+        hass.data[ANDROIDTV_DOMAIN][address] = device
+
+    if not hass.services.has_service(ANDROIDTV_DOMAIN, SERVICE_ADB_COMMAND):
+
+        def service_adb_command(service):
+            """Dispatch service calls to target entities."""
+            cmd = service.data.get(ATTR_COMMAND)
+            entity_id = service.data.get(ATTR_ENTITY_ID)
+            target_devices = [
+                dev
+                for dev in hass.data[ANDROIDTV_DOMAIN].values()
+                if dev.entity_id in entity_id
+            ]
+
+            for target_device in target_devices:
+                output = target_device.adb_command(cmd)
+
+                # log the output, if there is any
+                if output:
+                    _LOGGER.info(
+                        "Output of command '%s' from '%s': %s",
+                        cmd,
+                        target_device.entity_id,
+                        output,
+                    )
+
+        hass.services.register(
+            ANDROIDTV_DOMAIN,
+            SERVICE_ADB_COMMAND,
+            service_adb_command,
+            schema=SERVICE_ADB_COMMAND_SCHEMA,
+        )
+
+    if not hass.services.has_service(ANDROIDTV_DOMAIN, SERVICE_ADB_FILESYNC):
+
+        def service_adb_filesync(service):
+            """Transfer a file between your HA instance and an Android TV / Fire TV device."""
+            direction = service.data.get(ATTR_DIRECTION)
+            device_path = service.data.get(ATTR_DEVICE_PATH)
+            local_path = service.data.get(ATTR_LOCAL_PATH)
+            entity_id = service.data.get(ATTR_ENTITY_ID)
+            target_devices = [
+                dev
+                for dev in hass.data[ANDROIDTV_DOMAIN].values()
+                if dev.entity_id in entity_id
+            ]
 
-    hass.services.register(
-        ANDROIDTV_DOMAIN,
-        SERVICE_ADB_COMMAND,
-        service_adb_command,
-        schema=SERVICE_ADB_COMMAND_SCHEMA,
-    )
+            for target_device in target_devices:
+                target_device.adb_filesync(direction, local_path, device_path)
```

Comment:
You need to verify the local path is whitelisted with `hass.config.is_allowed_path()`

Reply:
Fixed in https://github.com/home-assistant/home-assistant/pull/30086/commits/16835bf67cda69fdead58f9a8c092fa4ba5249c2